### ISSUE TITLE
Spaces in accessorKey breaks table silently in layoutMode="grid"

### ISSUE THREAD

**Issue by MarkusEllyton** [KevinVandy/material-react-table#362]
### material-react-table version

1.7.1

### react & react-dom versions

v18.2.0

### Describe the bug and the steps to reproduce it

When spaces are used in accessorKey

```typescript
{
    header: 'First Name',
    accessorKey: 'first Name',
},
```
the custom CSS properties created to control width, min-width, flex, and other properties become invalid. The table rendering is completely broken when layoutMode is `"grid"` due to this, as cells end up with different widths on different rows. Any columns with spaces in accessorKey have flex-grow=0, making their width only depend on the content in a cell.

### Minimal, Reproducible Example - (Optional, but Recommended)

https://stackblitz.com/edit/github-1yw1sk?file=src/TS.tsx

### Screenshots or Videos (Optional)

The column sizing becomes inconsistent:
![image](https://user-images.githubusercontent.com/70698952/217779262-49665634-9937-4974-afcd-01152f41108d.png)

The CSS custom properties become invalid when using spaces:
![image](https://user-images.githubusercontent.com/70698952/217779477-cb840e35-47f4-42aa-9bda-9d0f30b1b14f.png)

### Do you intend to try to help solve this bug with your own PR?

None

### Terms

- [X] I understand that if my bug cannot be reliably reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by KevinVandy** [comment]
ah interesting. related to the new css-vars for column widths based off of the column.id. I'll push up a fix soon

**Comment by KevinVandy** [comment]
Thanks for reporting quickly, fixed in v1.7.2 - https://stackblitz.com/edit/github-1yw1sk-xgmamg?file=package.json

**Comment by oliverfrost1** [comment]
@KevinVandy This also happens for the "+" sign, are you able to fix this as well? https://stackblitz.com/edit/github-1yw1sk-fnzchg?file=src/TS.tsx
